src/main/scala/com/sky/kafka/topicloader/config/config.scala

```diff
@@ -0,0 +1,56 @@
+package com.sky.kafka.topicloader
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+import cats.data.ValidatedNec
+import cats.implicits._
+import com.typesafe.config.{Config => TypesafeConfig, ConfigException}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.util.{Failure, Success, Try}
+
+package object config {
+  type ValidationResult[A] = ValidatedNec[Throwable, A]
+
+  implicit class TryOps[A](t: Try[A]) {
+    def validate(): ValidationResult[A] = t.toEither.toValidatedNec
+  }
+
+  final case class PosInt(private val _value: Int) {
```

Comment:
can the default apply that is generated still be used to create an instance of this, by passing the validation from the apply below? you used to have to use a sealed abstract class to achieve that, but i dunno if they fixed that in the language or not

Reply:
I think you can do something like `Class PosInt Private` or `sealed abstract Class PosInt`, but what do you mean pass the validation?